### PR TITLE
[execution] removed unused output_capture

### PR DESCRIPTION
**Pull request by alangenfeld**
It looks like we erroneously set output_capture in this code path despite it not being used / passed forward. Remove it.

## How I Tested These Changes

updated test

## Changelog

- [bugfix] execute_job / execute_run with in_process execution should no longer retains unused references to outputs, allowing them to be garbage collected

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#25044** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25044?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster_tests/execution_tests/dynamic_tests/test_dynamic_output.py [hunk 1]
```diff
@@ -263,6 +264,7 @@ def test_dealloc_prev_outputs():
         with execute_job(
             reconstructable(no_leaks_plz),
             instance=inst,
+            run_config={"execution": {"config": {executor: {}}}},
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/execution_tests/dynamic_tests/test_dynamic_output.py:267]: why is this necessary? Maybe a comment

**Review by dpeng817 — APPROVED** [review]
